### PR TITLE
Be able to provide a custom router using the `router` setting

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

File: docs/source/developer/router.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+``python
+from guillotina.interfaces import IRequest
+from guillotina.traversal import TraversalRouter
+
+
+class MyRouter(TraversalRouter):
+    async def traverse(self, request: IRequest) -> IResource:
+        '''custom traversal here...'''
+        return await super().traverse(request)
```
Comment (tomgross) [inline docs/source/developer/router.md]: I would put a more substantial example here like in #168

**Comment by tomgross** [comment]
I was able to test this PR (rebased on master) successfully with my experimental link checker application written on top of guillotina: https://github.com/tomgross/LinkSpider

**Comment by vangheem** [comment]
Great! I'll rebase, update the docs and merge. Thank you!

**Comment by tomgross** [comment]
Thanks @vangheem !

**Pull request merged**